### ISSUE TITLE
Wrong loading substate is entered.

### ISSUE THREAD

**Issue by kellyselden** [ember-engines/ember-engines#433]
Given

```js
Router.map(function() {
  this.mount('my-engine');
});
```

and
`app/templates/loading.hbs`
`app/templates/my-engine-loading.hbs`

Only loading.hbs renders while waiting for my-engine. Shouldn't it render my-engine-loading.hbs?

**Comment by knownasilya** [comment]
wouldn't it be `app/tempaltes/my-engine/loading.hbs`?

**Comment by kellyselden** [comment]
@knownasilya Not according to https://guides.emberjs.com/v2.13.0/routing/loading-and-error-substates/

> It's important to note that for slow-model itself, Ember will not try to find a slow-model.loading template

**Comment by kellyselden** [comment]
I should also note that when I turn the `this.mount` to `this.route`, it loads the correct template `app/templates/my-engine-loading.hbs`.

**Comment by trentmwillis** [comment]
@kellyselden seems like a legit bug. The loading/error substates/routes have been a mess for a long time. We tried cleaning them up last year but looks like we missed this edge case.

I'd recommend adding a failing test [here](https://github.com/emberjs/ember.js/blob/36bd4f711b366fd7ef77ace0ac3ce816dc9a6c31/packages/ember-glimmer/tests/integration/application/engine-test.js#L449-L583) and (if you feel up to it) looking into fixing (which will probably be around [here](https://github.com/emberjs/ember.js/blob/299c1ab3d804fd92b0ec9b9c2e7a1fa100335753/packages/ember-routing/lib/system/router.js#L1189-L1202)).

Even though the changes will be upstream, we can keep this issue for tracking.

**Comment by nathanhammond** [comment]
This would need to be `application-loading` as part of the engine and hoisted from the engine to it's host. It can't load asynchronously and this was an _active_ design decision we made. (Don't know where the documentation for that decision is anymore.) It definitely does not live outside of the engine's namespace as you have described, @kellyselden.

Try it inside of the engine namespace to see if we fully accounted for it or didn't finish implementation.

**Comment by trentmwillis** [comment]
I don't remember the original decision making process, but I believe it was wrong. A host application should be able to provide a custom loading substate for when an Engine is loading.

**Comment by kellyselden** [comment]
I agree with @trentmwillis. It seems like it would make sense for an engine consumer to not depend on an engine to handle the loading substate, and to provide a "fallback" of sorts. It would also put an engine more in line with how routes work, which is a good thing IMO.

**Comment by jbailey4** [comment]
I also ran into related issues/confusion on loading sub-states, but eventually got it working after ready @nathanhammond comment. Using `application-loading.hbs` within the engine, itself works. Following the guides, it seems you can have `loading.hbs` or `application-loading.hbs`, but `loading.hbs` does not work in an engine it seems.

Also, 

> A host application should be able to provide a custom loading sub-state for when an Engine is loading.

My first inclination was to ONLY have the engine provide its own loading states, but I could see both ways though. 
Is the goal to keep allowing engines to provide a loading sub-state, and provide a way for the host application to "override" the engine's sub-state, either automatically or through some config?

**Comment by TigressBailey** [comment]
Just want to confirm whether this issue is fixed.
